autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+# -*- encoding: utf-8 -*-
+
+from typing import Union, Tuple
+import warnings
+
+import numpy as np
+import pandas as pd
+
+import sklearn.utils
+from sklearn import preprocessing
+from sklearn.compose import make_column_transformer
+
+import scipy.sparse
+
+
+class InputValidator:
+    """
+    Makes sure the input data complies with Auto-sklearn requirements.
+    Categorical inputs are encoded via a Label Encoder, if the input
+    is a dataframe.
+
+    This class also perform checks for data integrity and flags the user
+    via informative errors.
+    """
+    def __init__(
+        self,
+        feature_encoder_type: str = 'OrdinalEncoder',
+        target_encoder_type: str = 'LabelEncoder',
+    ):
+        self.valid_pd_enc_dtypes = ['category', 'bool']
+
+        # Whether we should treat the feature as a
+        # categorical or numerical input. If None,
+        # (The input was not a dataframe)
+        # The estimator will try to guess
```

Comment:
Doesn't it rely on information given by the user?

Reply:
Please don't forget to update this text.

Reply:
Updated it to be:

   # If a dataframe was provided, we populate
         # this attribute with the column types from the dataframe
         # That is, this attribute contains whether autosklearn
         # should treat a column as categorical or numerical
         # During fit, if the user provided feature_types, the user
         # constrain is honored. If not, this attribute is used.